backend/tests/test_course_build_processor.py

```diff
@@ -0,0 +1,478 @@
+        course_id=None,
+        course_enrollment_url=None,
+        course_enrollment_key=None,
+        school_district_id=school_district_id,
+        academic_year="AY 2024",
+        academic_year_short="AY24",
+        base_course_id=70,
+        status="completed",
+        creator_id=user_id,
+    )
+    db.add(course_build)
+    db.add(course_build2)
+    db.add(course_build3)
+    db.commit()
+
+
+def test_course_build_processor(mocker, db, create_course_builds):
+    sqs_client = boto3.client("sqs", region_name="azeroth")
+    sqs_stubber = botocore.stub.Stubber(sqs_client)
+
+    course_builds = db.query(CourseBuild).all()
+
+    initial_course_build = course_builds[0]
+
+    mock_sqs_data = {"course_build_id": initial_course_build.id}
+
+    mock_settings = mocker.Mock()
+    setattr(mock_settings, "SQS_QUEUE", "testqueue")
+    setattr(mock_settings, "SQS_POLL_INTERVAL_MINS", "1")
+    mocker.patch(
+        "rope.api.processors.course_build_processor.settings",
+        mock_settings,
+    )
+
+    engine = create_engine("postgresql://pguser:pgpassword@localhost/ropedb")
+    mock_factory = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+
+    mocker.patch(
+        "rope.api.processors.course_build_processor.get_db",
+        lambda: mock_factory,
+    )
+
+    mocker.patch(
+        "rope.api.processors.course_build_processor.get_moodle_user_role_by_shortname",
+        side_effects=["teacher", "student"],
+    )
+    mocker.patch(
+        "rope.api.routers.moodle.moodle_client.get_user_by_email",
+        return_value={
+            "id": 1,
+        },
+    )
+    mocker.patch(
+        "rope.api.processors.course_build_processor.create_course",
+        return_value={
+            "course_id": 77,
+            "course_enrolment_url": "https://enrolmenturl.com",
+            "course_enrolment_key": "amazing_enrolmentkey77",
+        },
+    )
+
+    sqs_stubber.add_response(
+        "get_queue_url",
+        {"QueueUrl": "https://testqueue"},
+        expected_params={"QueueName": "testqueue"},
+    )
+    sqs_stubber.add_response(
+        "receive_message",
+        {
+            "Messages": [
+                {
+                    "ReceiptHandle": "message1",
+                    "Body": json.dumps(mock_sqs_data),
+                }
+            ]
+        },
+        expected_params={
+            "QueueUrl": "https://testqueue",
+            "MaxNumberOfMessages": 10,
+            "WaitTimeSeconds": 20,
+        },
+    )
+    sqs_stubber.add_response(
+        "delete_message",
+        {},
+        expected_params={
+            "QueueUrl": "https://testqueue",
+            "ReceiptHandle": "message1",
+        },
+    )
+
+    sqs_stubber.activate()
+    mocker_map = {"sqs": sqs_client}
+    mocker.patch("boto3.client", lambda client: mocker_map[client])
+    mocker.patch("sys.argv", [""])
+    course_build_processor.main()
+
+    get_sessionmaker = course_build_processor.get_db()
```

Comment:
I see -- it's because you have the object held in `initial_course_build` and need to refresh it. You also don't really need to query all or assert the length of 1 since a query on primary key should only return 1. I would just replace these lines with the following:

```python
    db.refresh(initial_course_build)

    assert initial_course_build.status == "completed"
    assert initial_course_build.course_id == 77
    assert initial_course_build.course_enrollment_key == "amazing_enrolmentkey77"
    assert (
        initial_course_build.course_enrollment_url == "https://enrolmenturl.com"
    )
```

Reply:
Interesting - I thought a new query would have refreshed the data in the table, but according to this [documentation](https://docs.sqlalchemy.org/en/20/orm/session_state_management.html#refreshing-expiring), it looks like I would have needed to use `.expire()` before my new query.

Updated.